src/main/java/apoc/schema/ExceptionUtil.java

```diff
@@ -0,0 +1,70 @@
+package apoc.schema;
+
+import org.neo4j.internal.kernel.api.exceptions.KernelException;
+import org.neo4j.kernel.api.exceptions.Status;
+
+public final class ExceptionUtil {
+
+    /**
+     * Returns the root cause of an exception.
+     *
+     * Copied from {@link org.neo4j.helpers.Exceptions#rootCause(Throwable)} due to deprecation.
+     *
+     * @param caughtException exception to find the root cause of.
+     * @return the root cause.
+     * @throws IllegalArgumentException if the provided exception is null.
+     */
+    public static Throwable rootCause(Throwable caughtException) {
+        if (null == caughtException) {
+            throw new IllegalArgumentException("Cannot obtain rootCause from (null)");
+        }
+        Throwable root = caughtException;
+        while (root.getCause() != null) {
+            root = root.getCause();
```

Comment:
cause might be the exception it self so this might end up in an endless loop.

Reply:
Sorry this was added by mistake. Copied across from algos